csrc/optimization/mark_aliases_prepare.cpp

```diff
@@ -36,7 +36,8 @@ void MarkAliasesPreparePass::runPass(Fusion* fusion) {
       continue;
     }
 
-    if (aliased_io->isFusionOutput() && !aliased_io->isFusionInput() &&
+    if (tv->isFusionOutput() && aliased_io->isFusionOutput() &&
```

Comment:
QQ: what's the case here with (the root of `tv`'s alias is aliasing another tensor)
```
TensorView* aliased_io = analysis.getNearestAliasedIo(tv);
// aliased_io != nullptr
// and
analysis.getNearestAliasedIo(aliased_io) != nullptr;
```

Is it just avoiding segmenting when we have fusion outputs aliasing another output that could be just turned into an aliased inputs?

I'm getting curious here on how is concat going to play here. IIUC. concat would somehow change the aliasing direction, we'll be aliasing concat input to outputs.

Reply:
> Is it just avoiding segmenting when we have fusion outputs aliasing another output that could be just turned into an aliased inputs?

I'm happy to revisit this. My intention was to avoid too many no-op segments, e.g., https://github.com/NVIDIA/Fuser/blob/ee866bdf567b48b8b28da466cde7d75f93650d44/test/test_alias.cpp#L888-L889. But no-op segments become empty kernels after all, so might not be expensive at all. 

> I'm getting curious here on how is concat going to play here. IIUC. concat would somehow change the aliasing direction, we'll be aliasing concat input to outputs.

My plan is to put segment boundaries both before (which we already do) and after a concat.

Reply:
Got'ya. Thanks for pointing me to the example~